From b9fa2963cf12e6987b3a0acc219c4fa591e41be8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 3 Feb 2016 10:30:43 +0000
Subject: fix gcc build error

error: invalid operands of types '__gnu_cxx::__enable_if<true, double>::__type
{aka double}' and 'int' to binary 'operator%'

apparently there's a template returning double for abs for the
non-int/long/float/double argument case. So promote earlier to int so the
abs<int> is called

Change-Id: I882a27c5ec349f894c1c9f4857687360a46b55ae

diff --git a/sax/source/tools/converter.cxx b/sax/source/tools/converter.cxx
index 744415e..cf061c9 100644
--- a/sax/source/tools/converter.cxx
+++ b/sax/source/tools/converter.cxx
@@ -1250,7 +1250,7 @@ bool Converter::convertDuration(util::Duration& rDuration,
 
 
 static void
-lcl_AppendTimezone(OUStringBuffer & i_rBuffer, sal_Int16 const nOffset)
+lcl_AppendTimezone(OUStringBuffer & i_rBuffer, int const nOffset)
 {
     if (0 == nOffset)
     {
@@ -1435,7 +1435,7 @@ lcl_MaxDaysPerMonth(const sal_Int32 nMonth, const sal_Int32 nYear)
 static void lcl_ConvertToUTC(
         sal_Int16 & o_rYear, sal_uInt16 & o_rMonth, sal_uInt16 & o_rDay,
         sal_uInt16 & o_rHours, sal_uInt16 & o_rMinutes,
-        sal_Int16 const nSourceOffset)
+        int const nSourceOffset)
 {
     sal_Int16 nOffsetHours(abs(nSourceOffset) / 60);
     sal_Int16 const nOffsetMinutes(abs(nSourceOffset) % 60);
-- 
cgit v0.10.2

